lints/lint_ext_san_dns_name_underscore.go

```diff
@@ -0,0 +1,39 @@
+// lint_ext_san_dns_name_underscore.go
+
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"strings"
+)
+
+type SANDNSContainsUnderscore struct {
+	// Internal data here
+}
+
+func (l *SANDNSContainsUnderscore) Initialize() error {
+	return nil
+}
+
+func (l *SANDNSContainsUnderscore) CheckApplies(c *x509.Certificate) bool {
+	return util.IsExtInCert(c, util.SANOID)
+}
+
+func (l *SANDNSContainsUnderscore) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	for _, dns := range c.DNSNames {
+		if strings.Contains(dns, "_") {
```

Comment:
this probably came from here: https://github.com/awslabs/certlint/blob/master/lib/certlint/generalnames.rb line 187, don't know why it's in there as a warning though